src/api/Base.js

```diff
@@ -123,6 +134,78 @@ class Base {
     getCache(): Cache {
         return this.cache;
     }
+
+    /**
+     * Generic success handler
+     *
+     * @param {Object} data the response data
+     * @param {Function} successCallback the success callback
+     */
+    successHandler(data: Object): void {
+        if (!this.isDestroyed() && this.successCallback && typeof this.successCallback === 'function') {
+            this.successCallback(data);
+        }
+    }
+
+    /**
+     * Generic error handler
+     *
+     * @param {Object} data the response data
+     * @param {Function} errorCallback the error callback
+     */
+    errorHandler(error: Object): void {
+        if (!this.isDestroyed() && this.errorCallback && typeof this.errorCallback === 'function') {
+            this.errorCallback(error);
+        }
+    }
+
+    /**
+     * Generic GET request
+     *
+     * @param {string} id the file id
+     * @param {Object} params the query params
+     */
+    getData(id: string, params?: Object): Promise<Object> {
+        return this.xhr.get({
+            id: getTypedFileId(id),
+            url: this.getUrl(id),
+            params
+        });
+    }
+
+    /**
+     * Gets the URL for the API, meant to be overridden
+     * @param {string} id the file id
+     */
+    /* eslint-disable no-unused-vars */
+    getUrl(id: string) {
+        /* eslint-enable no-unused-vars */
+        throw new Error('Implement me!');
+    }
+
+    /**
+     * Generic API get
+     *
+     * @param {string} id the file id
+     * @param {Function} successCallback the success callback
+     * @param {Function} errorCallback the error callback
+     */
+    async get(id: string, successCallback: Function, errorCallback: Function): Promise<void> {
```

Comment:
Doesn't matter here, but probably not use keywords. get / set are js keywords.

Reply:
Hmm yeah there are get and set methods which you can have in classes. I wasnt able to find any documentation for using get and set as method names, nor does eslint complain. For future reference, can you point me to the documentation where you saw this?